### PR TITLE
BaSP-TG-8: TimeSheet

### PR DESCRIPTION
**Pull request by FiolAxel**
- Create Model for Timesheet
- GetAll
- GetByID

_I also worked in the Update function but it's in Ariana's PR_

### CODE REVIEW

File: src/controllers/time-sheets.js [hunk 1]
```diff
@@ -1,78 +1,37 @@
-const express = require('express');
-const fs = require('fs');
-const dataTimeSheets = require('../data/time-sheets.json');
+const Timesheets = require('../models/Time-sheets');
```
Comment (cocolobos) [inline src/controllers/time-sheets.js:1]: const should start with camelCase not CapitalCase

File: src/controllers/time-sheets.js [hunk 2]
```diff
@@ -1,78 +1,37 @@
...
-    dataTimeSheets.push(tsData);
-    fs.writeFile('src/data/time-sheets.json', JSON.stringify(dataTimeSheets), (err) => {
-      if (err) {
-        res.status(404).send(err);
-      } else {
-        res.status(201).json(tsData);
-      }
+const getByOne = async (req, res) => {
+  try {
+    const OneTimeSheet = await Timesheets.find({ _id: req.params.id });
```
Comment (cocolobos) [inline src/controllers/time-sheets.js:5]: same here

File: src/controllers/time-sheets.js [hunk 3]
```diff
@@ -1,78 +1,37 @@
...
-            res.send(err);
-          } else {
-            res.send('Updated correctly');
-          }
-        });
-        res.json({ msg: 'Timesheet update', updateTs });
-      }
+const getAll = async (req, res) => {
+  try {
+    const AllTimeSheets = await Timesheets.find({});
```
Comment (cocolobos) [inline src/controllers/time-sheets.js:22]: same here

File: src/models/Time-sheets.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  description: String,
+  date: Date,
+  task: String,
+  validate: Boolean,
+  projectId: String,
+  employee: {
+    name: String,
+    role: {
+      type: String,
+      enum: ['DEV', 'TL', 'PM', 'QA'],
```
Comment (cocolobos) [inline src/models/Time-sheets.js:13]: some fields should be required

**Review by EzeRocco — APPROVED** [review]

File: src/models/Time-sheets.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    },
+  },
+});
+
```
Comment (lucasrendo) [inline src/models/Time-sheets.js:27]: It could have more validations besides the type of the field

File: src/routes/index.js [hunk 6]
```diff
@@ -2,10 +2,14 @@ import express from 'express';
...
 // Routes import
 import tasksRouter from './tasks';
+import timeSheetsRouter from './time-sheets';
 
 const router = express.Router();
 
 router
   .use('/tasks', tasksRouter);
 
+router
```
Comment (lucasrendo) [inline src/routes/index.js:12]: you can use the router on line 9 by chaining methods. No need to call the router again

**Review by lucasrendo — APPROVED** [review]

**Pull request merged**